### DISCUSSION TITLE
Output Details of Exported Model

### DISCUSSION THREAD

**Discussion by Mervetemurr** [open-edge-platform/anomalib#2312] [Q&A] [answered]
Hello,

I have trained the Patchcore and Padim models with my custom dataset and exported as onnx file format.
I will use the onnx file on mobile by using Onnxruntime Mobile API.
For Patchcore model my onnx file has 2 output 
1. tensor: float32[1,1,224,224] - I assumed that this my a heatmap data
2.tensor: float32[1] - I assumed that this is prediction score data
I don't have the information of classification data, is result normal or abnormal?
Could you please inform me about output layers? 

dataset_root = "C:/Users/m.temur/anomalib/dataset/sponge"
datamodule = Folder(
    name="images_sponge",
    root=dataset_root,
    normal_dir="normal",
    abnormal_dir="abnormal",
    task=TaskType.CLASSIFICATION,
![onnx properties](https://github.com/user-attachments/assets/ea863f95-84fe-4ef1-a749-217089b18fa6)

    seed=42,
    normal_split_ratio=0.2, # default value
    val_split_mode=ValSplitMode.FROM_TEST, # default value
    val_split_ratio=0.5, # default value
    train_batch_size=8, # default value
    eval_batch_size=8, # default value
)

Exporting model to Onnx steps:
input_size  = model.input_size
print(model.pre)
print(input_size)
export_model = engine.export(
    model=model,
    input_size=input_size,
    export_type=ExportType.ONNX,
    export_root=str("C:/Users/m.temur/anomalib/exported_onnx/padim/"),
    ckpt_path= None
)

**Comment by blaz-r** [comment, marked as answer]
I believe that the model exported by Engine is what's contained inside `torch_model` of specific architecture. So for PatchCore you can see the forward code inside `torch_model` here:
https://github.com/openvinotoolkit/anomalib/blob/f473df85e54e72824cba028a04fc210c57c4c30b/src/anomalib/models/image/patchcore/torch_model.py#L104-L106

So your assumption about returned values is correct, one is anomaly map, the other is anomaly score. Some models like [padim](https://github.com/openvinotoolkit/anomalib/blob/f473df85e54e72824cba028a04fc210c57c4c30b/src/anomalib/models/image/padim/torch_model.py#L143-L149) only return anomaly map, in which case the anomaly score is calculated as `max(anomaly_map)`.

Since you want to determine whether the score indicates normal or anomalous image, you'll need to threshold the anomaly score somehow. Anomalib calculates the threshold which yields the best F1 score on val set (MaxF1) so you can use that threshold which should be saved inside the `metadata.json`, next to .onnx file.

Also check out the [OpenVino inferene python script](https://github.com/openvinotoolkit/anomalib/blob/main/tools/inference/openvino_inference.py) (that also covers onnx inference), and the OpenVino inferencer class.
